### PR TITLE
Play system soundIDs directly

### PR DESCRIPTION
**Pull request by lastObject**
First step in completing #30

### CODE REVIEW

File: Example/Example/Base.lproj/Main.storyboard [hunk 1]
```diff
@@ -132,9 +133,4 @@
             <point key="canvasLocation" x="270" y="393"/>
         </scene>
     </scenes>
-    <simulatedMetricsContainer key="defaultSimulatedMetrics">
-        <simulatedStatusBarMetrics key="statusBar"/>
-        <simulatedOrientationMetrics key="orientation"/>
-        <simulatedScreenMetrics key="destination" type="retina4"/>
-    </simulatedMetricsContainer>
 </document>
```
Comment (jessesquires) [inline Example/Example/Base.lproj/Main.storyboard]: let's revert all of these unrelated changes

File: Source/JSQSystemSoundPlayer.h [hunk 2]
```diff
@@ -159,6 +160,14 @@ typedef void(^JSQSystemSoundPlayerCompletionBlock)(void);
...
                      fileExtension:(NSString *)fileExtension
                         completion:(nullable JSQSystemSoundPlayerCompletionBlock)completionBlock;
+/**
+ *  play predefined system sound
+ *
+ *  @param soundID         refer http://iphonedevwiki.net/index.php/AudioServices
+ *  @param completionBlock A block called after the sound has stopped playing.
+ */
+- (void)playPredefinedSystemSoundWithSoundID:(SystemSoundID)soundID
+                             completionBlock:(nullable JSQSystemSoundPlayerCompletionBlock)completionBlock;
```
Comment (jessesquires) [inline Source/JSQSystemSoundPlayer.h]: let's rename this too:

```objective-c
- (void)playSoundForSoundID:(SystemSoundID)soundID 
                 completion:(nullable JSQSystemSoundPlayerCompletionBlock)completionBlock;
```

File: Source/JSQSystemSoundPlayer.h [hunk 3]
```diff
@@ -159,6 +160,14 @@ typedef void(^JSQSystemSoundPlayerCompletionBlock)(void);
 - (void)playAlertSoundWithFilename:(NSString *)filename
                      fileExtension:(NSString *)fileExtension
                         completion:(nullable JSQSystemSoundPlayerCompletionBlock)completionBlock;
+/**
+ *  play predefined system sound
+ *
+ *  @param soundID         refer http://iphonedevwiki.net/index.php/AudioServices
```
Comment (jessesquires) [inline Source/JSQSystemSoundPlayer.h]:
```
 @param soundID        The SystemSoundID to play.
```

File: Source/JSQSystemSoundPlayer.h [hunk 4]
```diff
@@ -159,6 +160,14 @@ typedef void(^JSQSystemSoundPlayerCompletionBlock)(void);
 - (void)playAlertSoundWithFilename:(NSString *)filename
                      fileExtension:(NSString *)fileExtension
                         completion:(nullable JSQSystemSoundPlayerCompletionBlock)completionBlock;
+/**
+ *  play predefined system sound
```
Comment (jessesquires) [inline Source/JSQSystemSoundPlayer.h]: change to:

```
Plays the sound that corresponds to the specified `SystemSoundID`.
```

File: Source/JSQSystemSoundPlayer.m [hunk 5]
```diff
@@ -122,39 +120,30 @@ - (void)setBundle:(NSBundle *)bundle
...
-
     if (!self.on) {
         return;
     }
 
-    if (![self.sounds objectForKey:filename]) {
-        [self addSoundIDForAudioFileWithName:filename extension:extension];
-    }
-
-    SystemSoundID soundID = [self soundIDForFilename:filename];
```
Comment (jessesquires) [inline Source/JSQSystemSoundPlayer.m]: the re-ordering here makes this diff a bit hard to read. let's leave everything above this line how it was, then extract what's below this line into the new method, `playSoundForSoundID: asAlert: completion:`

File: Source/JSQSystemSoundPlayer.h [hunk 6]
```diff
@@ -159,6 +160,14 @@ typedef void(^JSQSystemSoundPlayerCompletionBlock)(void);
...
+ *
+ *  @param soundID         refer http://iphonedevwiki.net/index.php/AudioServices
+ *  @param completionBlock A block called after the sound has stopped playing.
+ */
+- (void)playPredefinedSystemSoundWithSoundID:(SystemSoundID)soundID
+                             completionBlock:(nullable JSQSystemSoundPlayerCompletionBlock)completionBlock;
```
Comment (jessesquires) [inline Source/JSQSystemSoundPlayer.h]: Oh, we'll also need unit tests for this new method :smile:

**Comment by jessesquires** [comment]
Thanks for working on this @lastObject ! :smile: 

Looks good! :+1:  I've left a few comments to address, then I'm happy to merge!

Also, could you squash your commits into 1?

**Comment by lastObject** [comment]
@jessesquires 
thanks for your fantastic work, just too newbie to git [](https://assets-cdn.github.com/images/icons/emoji/unicode/1f60a.png) try to finish it on weekend,

**Comment by jessesquires** [comment]
@lastObject - no worries! :smile: there's no rush. let me know if you need any help

**Comment by lastObject** [comment]
hi @jessesquires,  review my pr plz. let me know if needs any change. cheers

File: Source/JSQSystemSoundPlayer.m [hunk 7]
```diff
@@ -130,31 +128,41 @@ - (void)playSoundWithName:(NSString *)filename
     NSParameterAssert(filename != nil);
     NSParameterAssert(extension != nil);
 
-    if (!self.on) {
-        return;
-    }
```
Comment (jessesquires) [inline Source/JSQSystemSoundPlayer.m]: let's put this back, so we can bail as early as possible.

File: Source/JSQSystemSoundPlayer.m [hunk 8]
```diff
@@ -130,31 +128,41 @@ - (void)playSoundWithName:(NSString *)filename
...
+}
+
+- (void)playSoundWithSoundID:(SystemSoundID)soundID
+                     isAlert:(BOOL)isAlert
+                  completion:(nullable JSQSystemSoundPlayerCompletionBlock)completionBlock
+{
+    if (!self.on) {
+        return;
+    }
+
```
Comment (jessesquires) [inline Source/JSQSystemSoundPlayer.m]: 👍

**Comment by jessesquires** [comment]
nice! thanks @lastObject 😄 

- 1 more comment
- it looks like `Main.storyboard` was changed. it's hard to review. can you post a screenshot of the view?

**Comment by jessesquires** [comment]
✨ ping ✨ @lastObject 😄

File: Source/JSQSystemSoundPlayer.m (hunk 7 above)
Comment (lastObject) [inline Source/JSQSystemSoundPlayer.m]: actuary i was so struggling with here, i will put it back

**Comment by lastObject** [comment]
simulator screen shot apr 22 2016 9 19 07 am

this is new page looks like

File: Source/JSQSystemSoundPlayer.m (hunk 7 above)
Comment (lastObject) [inline Source/JSQSystemSoundPlayer.m]: @jessesquires 
now it will check ```self.on```  -twice- if you play a normal file, want to extract a new method, but struggling on method name... have a good idea?
cheers

**Comment by jessesquires** [comment]
Thanks @lastObject ! 👍 

I think we're good to go. I'll merge this soon! 🎉
